src/flamegraph/color/palettes.rs

```diff
@@ -59,6 +59,21 @@ pub(super) mod perl {
     }
 }
 
+pub(super) mod python {
+    use crate::flamegraph::color::BasicPalette;
+
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.starts_with("native@") { // austin-specific format for native calls
+            return BasicPalette::Aqua;
+        } else if name.contains("site-packages/") {
+            return BasicPalette::Yellow;
+        } else if name.contains("/python") || name.starts_with("<frozen importlib") { // stdlib
```

Comment:
This feels quite brittle. For example, if I do all my Python development under
```
~/dev/python/
```
then _every_ Python file I have will be considered part of the Python standard library. Is there a pattern with lower risk of false positives we can use here?

Reply:
You're right. I don't think there is going to be a perfect way of doing it but we can probably do better than this. I've done a couple more experiments:

On a standard Python installation on Windows the stdlib lives in e.g.:
```
C:\Users\Mike\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py
```

On a standard Ubuntu installation it lives in e.g.:
```
/usr/lib/python3.9/concurrent/futures/thread.py
```

Based on [this](https://stackoverflow.com/questions/2927993/where-are-the-python-modules-stored) SO thread this looks fairly standard, though given the number of ways there are to install Python I would be surprised if it's universal.

So I think a reasonable way of identifying stdlib files would be:
```
name.contains("/python") && name.contains("/lib/")
```

This would need to come after the check for `"/site-packages/"` since those also live under `/lib/python3.9`..

We also need to handle backslashes for windows traces. Would you prefer to just replace all `\` with `/` at the start of this function or duplicate each condition? Or is there a smart way to conditionally compile the function based on the OS?

Reply:
I've updated this to look for `/lib/` as well, and also added equivalent checks for Windows paths

Reply:
Here's another proposal:
```rust
if Path::new(name).iter().any(|c| c.strip_prefix("python").map_or(false, |version| version.chars().all(|c| c.is_digit() || c == '.')))
```
It's not beautiful, but I think it's unlikely to produce false positives, and I _think_ it should map onto both the OS examples you gave above?

Reply:
Ah, and as far as `/` vs `\` is concerned, use https://doc.rust-lang.org/std/path/constant.MAIN_SEPARATOR.html

Reply:
Or, actually, for that too you could use:
```rust
if Path::new(name).iter().any(|c| c == "site-packages")
```

Reply:
Very nice! I had to convert from `OsString` to `str` and added a `to_ascii_lowercase` to handle the uppercased path on Windows